snuba/clusters/cluster.py

```diff
@@ -96,14 +107,39 @@ def get_writer(
 class ClickhouseCluster(Cluster[SqlQuery, ClickhouseWriterOptions]):
     """
     ClickhouseCluster provides a reader, writer and Clickhouse connections that are
-    shared by all storages located on the cluster
+    shared by all storages located on the cluster.
+
+    ClickhouseCluster is initialized with a single address (host/port/http_port),
+    which is used for all read and write operations related to the cluster. This
+    address can refer to either the address of the actual ClickHouse server, or a
+    proxy server (e.g. for load balancing).
```

Comment:
Maybe add a couple of lines to explain the topology of the clickhosue cluster with regards to snuba: snuba nodes that run queries, clickhouse hosts that have dist tables that are hidden behind one address and clickhosue nodes that store data.